styx-client/src/main/java/com/spotify/styx/client/StyxApolloClient.java

```diff
@@ -255,7 +255,7 @@
         WorkflowId.create(componentId, workflowId),
         parameter);
     try {
-      final ByteString payload = serialize(Event.dequeue(workflowInstance));
+      final ByteString payload = serialize(Event.retryAfter(workflowInstance, 0L));
```

Comment:
How about introducing a more specific API for instructing styx to `retry` or `halt` a workflow instance instead of injecting raw events?

Reply:
Added